From 880e1698fac7436b601858816b57834c49a4ad1a Mon Sep 17 00:00:00 2001
From: Ashish Singhi <ashish.singhi@huawei.com>
Date: Thu, 2 Apr 2015 10:37:02 +0530
Subject: [PATCH 254/268] HBASE-11386 Backport HBASE-12601 to all active branches other than master

The correct JIRA nubmer is HBASE-13386, not HBASE-11386.

Reason: Supportability
Author: Ashish Singhi
Ref: CDH-31924
---
 hbase-shell/src/main/ruby/shell/commands/grant.rb  |    5 ++++-
 hbase-shell/src/main/ruby/shell/commands/revoke.rb |    9 ++++++++-
 .../main/ruby/shell/commands/user_permission.rb    |    4 ++++
 3 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/hbase-shell/src/main/ruby/shell/commands/grant.rb b/hbase-shell/src/main/ruby/shell/commands/grant.rb
index 0e8a65c..4a995ff 100644
--- a/hbase-shell/src/main/ruby/shell/commands/grant.rb
+++ b/hbase-shell/src/main/ruby/shell/commands/grant.rb
@@ -27,11 +27,14 @@ Syntax : grant <user> <permissions> [<@namespace> [<table> [<column family> [<co
 permissions is either zero or more letters from the set "RWXCA".
 READ('R'), WRITE('W'), EXEC('X'), CREATE('C'), ADMIN('A')
 
-Note: A namespace must always precede with '@' character.
+Note: Groups and users are granted access in the same way, but groups are prefixed with an '@' 
+      character. In the same way, tables and namespaces are specified, but namespaces are 
+      prefixed with an '@' character.
 
 For example:
 
     hbase> grant 'bobsmith', 'RWXCA'
+    hbase> grant '@admins', 'RWXCA'
     hbase> grant 'bobsmith', 'RWXCA', '@ns1'
     hbase> grant 'bobsmith', 'RW', 't1', 'f1', 'col1'
     hbase> grant 'bobsmith', 'RW', 'ns1:t1', 'f1', 'col1'
diff --git a/hbase-shell/src/main/ruby/shell/commands/revoke.rb b/hbase-shell/src/main/ruby/shell/commands/revoke.rb
index 57a2530..768989b 100644
--- a/hbase-shell/src/main/ruby/shell/commands/revoke.rb
+++ b/hbase-shell/src/main/ruby/shell/commands/revoke.rb
@@ -22,10 +22,17 @@ module Shell
       def help
         return <<-EOF
 Revoke a user's access rights.
-Syntax : revoke <user> [<table> [<column family> [<column qualifier>]]
+Syntax : revoke <user> [<@namespace> [<table> [<column family> [<column qualifier>]]]]
+
+Note: Groups and users access are revoked in the same way, but groups are prefixed with an '@' 
+      character. In the same way, tables and namespaces are specified, but namespaces are 
+      prefixed with an '@' character.
+
 For example:
 
     hbase> revoke 'bobsmith'
+    hbase> revoke '@admins'
+    hbase> revoke 'bobsmith', '@ns1'
     hbase> revoke 'bobsmith', 't1', 'f1', 'col1'
     hbase> revoke 'bobsmith', 'ns1:t1', 'f1', 'col1'
 EOF
diff --git a/hbase-shell/src/main/ruby/shell/commands/user_permission.rb b/hbase-shell/src/main/ruby/shell/commands/user_permission.rb
index 5d8bf8a..e4673fc 100644
--- a/hbase-shell/src/main/ruby/shell/commands/user_permission.rb
+++ b/hbase-shell/src/main/ruby/shell/commands/user_permission.rb
@@ -23,9 +23,13 @@ module Shell
         return <<-EOF
 Show all permissions for the particular user.
 Syntax : user_permission <table>
+
+Note: A namespace must always precede with '@' character.
+
 For example:
 
     hbase> user_permission
+    hbase> user_permission '@ns1'
     hbase> user_permission 'table1'
     hbase> user_permission 'namespace1:table1'
     hbase> user_permission '.*'
-- 
1.7.0.4

